### DISCUSSION TITLE
Is it possible to merge history from an older backup?

### DISCUSSION THREAD

**Discussion by mzso** [erosman/support#433] [Q&A]
Hi @erosman !

I'm thinking you're the best person to ask. [bug1753729](https://bugzilla.mozilla.org/show_bug.cgi?id=1753729#c13) recently destroyed most of my history. And I have a few weeks old backup.
Is there a way to import/merge older history items from that backup. Of course I have some new history since than and more importantly bookmarks and I don't want to lose those, so simply copying  places.sqlite is not such a good idea.

**Comment by erosman** [comment]
Hi

You can try the Restore (replace the file) or Import. I am not sure if the import merges the old data with the new, so you might want to back-up you data first.

- [Import data from another browser](https://support.mozilla.org/en-US/kb/import-data-another-browser)
- [Recovering important data from an old profile](https://support.mozilla.org/en-US/kb/recovering-important-data-from-an-old-profile)
- [Keep History format when exporting/importing - Mozilla Support](https://support.mozilla.org/en-US/questions/1279663)

**Reply by mzso** [reply]
Thanks.

**Comment by mzso** [comment]
[**@erosman**](https://github.com/mzso) commented on [2022. márc. 16. 06:10 CET](https://github.com/erosman/support/discussions/433#discussioncomment-2370041 "2022-03-16T05:10:50Z - Replied by Github Reply Comments"):
> Hi
> 
> You can try the Restore (replace the file) or Import. I am not sure if the import merges the old data with the new, so you might want to back-up you data first.
> 
> *   [Import data from another browser](https://support.mozilla.org/en-US/kb/import-data-another-browser)
> *   [Recovering important data from an old profile](https://support.mozilla.org/en-US/kb/recovering-important-data-from-an-old-profile)
> *   [Keep History format when exporting/importing - Mozilla Support](https://support.mozilla.org/en-US/questions/1279663)

Well, looking at it closely. Firefox can restore bookmarks from HTML and automatic backups. (Rather annoying that the "restore" menu doesn't actually specify what it tries to restore until it asks whether I want to overwrite current bookmarks with old ones....)
It can also import history from other browsers, but not Firefox... Ugh.
So I guess I have to find a way to merge the history tables from two different places.sqlite files. (maybe favicons.sqlite as well?)

**Comment by erosman** [comment]
You can also try asking in https://chat.mozilla.org/#/room/#nightly:mozilla.org
